File: Problem 127.py
import math
def find_prime_factors(n):
    factors = []
    i = 3
    while n > 1:
        if n%2 == 0:
            factors.append(2)
            n = n/2
        elif n%i == 0:
            factors.append(i)
            n = n/i
        else:
            i += 2
    return factors

def make_unique(arr):
    arr2 = []
    size = len(arr)
    for i in range(size):
        check = True
        for j in arr2:
            if j == min(arr):
                check = False
        if check == True:
            arr2.append(min(arr))
        del arr[arr.index(min(arr))]
    return arr2

def rad(n):
    arr = find_prime_factors(n)
    arr = make_unique(arr)
    p = 1
    for prime in arr:
        p *= prime
    return p

def GCD(a,b):
    while a != b:
        if a > b:
            a -= b
        else:
            b -= a
    return a

def is_abc_hit(a,b,c):
    if GCD(a,b) == 1:
        if GCD(a,c) == 1:
            if GCD(b,c) == 1:
                if rad(a)*rad(b)*rad(c) < c:
                    return True
    return False

def find_sum_c(limit):
    suma = 0
    for c in range(3, limit):
        for b in range(2,c):
            for a in range(1,b):
                if a+b == c:
                    if is_abc_hit(a,b,c):
                        suma += c
                        print(a,b,c)
    return suma

print(find_sum_c(100))

File: Problem 127.cpp
#include <iostream>
#include <math.h>
#include <vector>

using namespace std;

vector<unsigned long int> find_primes(int limit)
{
	vector<unsigned long int> primes;
	primes.push_back(2);
	bool isPrime;
	for(unsigned long int i = 3;i < limit;i += 2)
	{
		isPrime = true;
		for(unsigned long int j = 1; j<primes.size(); j++)
		{
			if (j > sqrt(i))
			{
				break;
			}
			if(i%primes[j] == 0)
			{
				isPrime = false;
				break;
			}
		}
		if(isPrime)
		{
			primes.push_back(i);
		}
	}
	return primes;
}
vector<unsigned long int> primes = find_primes(1000000);

vector<unsigned long int> find_prime_factors(unsigned long int n)
{
	vector<unsigned long int> factors;
	if (n == 1)
	{
		factors.push_back(1);
		return factors;
	}
	unsigned long int i = 0;
	while (n > 1)
	{
		if (n%primes[i] == 0)
		{
			factors.push_back(primes[i]);
			n = n/primes[i];
		}
		else
		{
			i += 1;
		}
	}
	return factors;
}

unsigned long int rad(unsigned long int n)
{
	vector<unsigned long int> arr = find_prime_factors(n);
	unsigned long int p = 1;
	if (n == 1)
	{
		return 1;
	}
	for (int i = 0; i < arr.size();i++)
	{
		if(p%arr[i] != 0)
		{
		    p *= arr[i];
	    }
	}
	return p;
}

unsigned long int GCD(unsigned long int a, unsigned long int b)
{
	if (a == 1)
	{
		return 1;
	}
	bool check = false;
	if (b-a < 500)
	{
		while (a != b)
		{
			if (a > b)
			{
				a -= b;
			}
			else
			{
				b -= a;
			}
			if (abs(b-a) > 500)
	    	{
	    		check = true;
		    	break;
	    	}
		}
		if (check == false)
		{
		    return a;
		}
	}
	int i = 0;
	while(primes[i] < sqrt(a)+3)
	{
		if(a%primes[i] == 0)
		{
			if (b%primes[i] == 0)
			{
				return 2;
			}
		}
		i++;
	}
	return 1;
}

vector<unsigned long int> radicalize(int limit)
{
	vector<unsigned long int> radicals;
    unsigned long int c;
     for(c = 1; c<= limit; c++)
     {
        radicals.push_back(rad(c));
     }
     return radicals;
}


int find_sum_c(int limit)
{
	vector<unsigned long int> radicals = radicalize(limit);
	cout << "Radicals initialized" << endl;
	unsigned long int suma = 0;
	unsigned long long int p,q,r;
	unsigned int c,a,b;
	for(c = 3; c < limit;c++)
	{
		p = radicals[c-1];
		if (p < c)
		{
			b = c-1;
		    a = 1;
		    while(a<b)
	    	{
	    		q = p*radicals[b-1]*radicals[a-1];
		        if (q < c)
		        {
		        	if (GCD(a,b) == 1 && GCD(a,c) == 1 && GCD(b,c) == 1)
	                {
			            suma += c;
		            }
	            }
		        b--;
		        a++;
	    	}
	    }
    }   
	return suma;
}

int main()
{
	cout << "Primes initialized" << endl;
	cout << find_sum_c(120000);
	
	
	return 0;
}
